File: crossecore-csharp/Java/Java/Java_/Modifier.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	
	public interface Modifier 
	: ASTNode
	{
		VisibilityKind visibility
		{
			get;
		set;
		}
		InheritanceKind inheritance
		{
			get;
		set;
		}
		bool static_
		{
			get;
		set;
		}
		bool transient
		{
			get;
		set;
		}
		bool volatile_
		{
			get;
		set;
		}
		bool native
		{
			get;
		set;
		}
		bool strictfp
		{
			get;
		set;
		}
		bool synchronized
		{
			get;
		set;
		}
		BodyDeclaration bodyDeclaration
		{
			get;
			set;
		}
		SingleVariableDeclaration singleVariableDeclaration
		{
			get;
			set;
		}
		VariableDeclarationStatement variableDeclarationStatement
		{
			get;
			set;
		}
		VariableDeclarationExpression variableDeclarationExpression
		{
			get;
			set;
		}
	}
}

File: crossecore-typescript/java_/Modifier.ts
import {Set} from "ecore/Set";
import {Bag} from "ecore/Bag";
import {VisibilityKind} from "./VisibilityKind";
import {InheritanceKind} from "./InheritanceKind";
import {SingleVariableDeclaration} from "./SingleVariableDeclaration";
import {VariableDeclarationExpression} from "./VariableDeclarationExpression";
import {ASTNode} from "./ASTNode";
import {VariableDeclarationStatement} from "./VariableDeclarationStatement";
import {BodyDeclaration} from "./BodyDeclaration";

export interface Modifier
extends ASTNode

{
	visibility:VisibilityKind;
	inheritance:InheritanceKind;
	static:boolean;
	transient:boolean;
	volatile:boolean;
	native:boolean;
	strictfp:boolean;
	synchronized:boolean;
	
	bodyDeclaration:BodyDeclaration;
	singleVariableDeclaration:SingleVariableDeclaration;
	variableDeclarationStatement:VariableDeclarationStatement;
	variableDeclarationExpression:VariableDeclarationExpression;
	

}

